client/src/components/AnchorSelectorField/AnchorSelectorField.js

```diff
@@ -16,140 +15,131 @@ import PropTypes from 'prop-types';
 
 const noop = () => null;
 
-class AnchorSelectorField extends SilverStripeComponent {
-  constructor(props) {
-    super(props);
-
-    this.handleChange = this.handleChange.bind(this);
-    this.handleLoadingError = this.handleLoadingError.bind(this);
-  }
-
-  componentDidMount() {
-    this.ensurePagesLoaded();
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.props.pageId !== prevProps.pageId) {
-      this.ensurePagesLoaded();
+const AnchorSelectorField = ({
+  extraClass = '',
+  name,
+  onChange,
+  value = '',
+  pageId,
+  anchors = [],
+  loadingState,
+  onLoadingError = noop,
+  data,
+  CreatableSelectComponent = CreatableSelect,
+  actions,
+}) => {
+  const handleLoadingError = (error) => {
+    if (onLoadingError === noop) {
+      throw error;
     }
-  }
+    // Custom error handling
+    return onLoadingError({
+      errors: [
+        {
+          value: error.message,
+          type: 'error',
+        },
+      ],
+    });
+  };
 
   /**
    * Lazy-triggers load of the dropdown based on pageId
    *
    * @param {Object} props - Props to check
    * @return {Promise} The promise object
    */
-  ensurePagesLoaded(props = this.props) {
+  const ensurePagesLoaded = () => {
```

Comment:
Have removed the weird `props = this.props` param which should never have been used